src/js/components/DataTable/Body.js

```diff
@@ -181,20 +178,25 @@ const Body = forwardRef(
 
     return (
       <Keyboard
-        onEnter={
-          // active is undefined if user never used the keyboard
-          onClickRow &&
-          active >= 0 &&
-          (!disabled ||
-            !disabled.includes(datumValue(data[active], primaryProperty)))
-            ? (event) => {
-                event.persist();
-                const adjustedEvent = event;
-                adjustedEvent.datum = data[active];
-                onClickRow(adjustedEvent);
-              }
-            : undefined
-        }
+        onEnter={(event) => {
+          if (
+            (onClickRow && active >= 0 && !disabled) ||
+            !disabled.includes(datumValue(data[active], primaryProperty))
+          ) {
+            if (typeof onClickRow === 'function') {
+              event.persist();
+              const adjustedEvent = event;
+              adjustedEvent.datum = data[active];
+              onClickRow(adjustedEvent);
+            }
+            if (onClickRow === 'onSelect') {
```

Comment:
```suggestion
            else if (onClickRow === 'onSelect') {
```
Fix formatting locally w linter

Reply:
done